interoperator/controllers/multiclusterdeploy/watchmanager/internal.go

```diff
@@ -101,9 +106,89 @@ func (wm *watchManager) addCluster(clusterID string) error {
 	return nil
 }
 
+func (wm *watchManager) requeueSFCRs(cachedClient kubernetes.Client, clusterID string) error {
+	ctx := context.TODO()
+	if wm.isWatchingOnSfcrRequeue(clusterID) {
+		log.Info("Already watching on instances in ", "clusterID", clusterID)
+		return nil
+	}
+	cw := wm.getClusterWatch(clusterID)
+	if cw == nil {
+		log.Info("Already watching on instances in ", "clusterID", clusterID)
+		return nil
+	}
+
+	sfserviceinstances := &osbv1alpha1.SFServiceInstanceList{}
+	instance_options := &kubernetes.ListOptions{}
+	kubernetes.MatchingLabels{"state": "delete"}.ApplyToList(instance_options)
+	kubernetes.MatchingFields{"spec.clusterId": "clusterID"}.ApplyToList(instance_options)
+	kubernetes.MatchingFields{"status.state": "in progress"}.ApplyToList(instance_options)
+
+	for more := true; more; more = (sfserviceinstances.Continue != "") {
+		err := cachedClient.List(ctx, sfserviceinstances, instance_options, kubernetes.Limit(constants.ListPaginationLimit),
+			kubernetes.Continue(sfserviceinstances.Continue))
+		if err != nil {
+			log.Error(err, "error while fetching sfserviceinstances")
+			return err
+		}
+		for _, sfserviceinstance := range sfserviceinstances.Items {
+			instance := sfserviceinstance.DeepCopy()
+			metaObject, err := meta.Accessor(instance)
+			if err != nil {
+				log.Error(err, "failed to process watch event for sfserviceinstance", "clusterID", cw.clusterID)
+				continue
+			}
+			cw.instanceEvents <- event.GenericEvent{
+				Meta:   metaObject,
+				Object: instance,
+			}
+		}
+	}
+
+	sfservicebindings := &osbv1alpha1.SFServiceBindingList{}
+	binding_options := &kubernetes.ListOptions{}
+	kubernetes.MatchingLabels{"state": "delete"}.ApplyToList(binding_options)
+	kubernetes.MatchingFields{"status.state": "in progress"}.ApplyToList(binding_options)
+
+	for moreBindings := true; moreBindings; moreBindings = (sfservicebindings.Continue != "") {
+		err := cachedClient.List(ctx, sfservicebindings, binding_options, kubernetes.Limit(constants.ListPaginationLimit),
+			kubernetes.Continue(sfservicebindings.Continue))
+		if err != nil {
+			log.Error(err, "error while fetching sfservicebindings")
+			return err
+		}
+		for _, sfservicebinding := range sfservicebindings.Items {
+			if sf_clusterID, err := sfservicebinding.GetClusterID(cachedClient); err == nil && sf_clusterID == clusterID {
```

Comment:
Hi @jintusebastian , we couldnt add the cluster id to list options above? like the way we did for sfserviceInstance

Reply:
ClusterID is not part of sfserviceBinding. So we need to get the clusterID from corresponding sfserviceInstance. 
`sfservicebinding.GetClusterID()` is helper method to get it.